### PR TITLE
Creates new room code before join for participants 

### LINKED ISSUE #13: Create individual rooms on Join
Currently, everyone joins one big public room. It would be better if the users join a new room with a random ID, that they can invite other people to it using the room code.

- [x] Create a new room on join [using Rooms API](https://www.100ms.live/docs/server-side/v2/api-reference/Rooms/create-via-api)
- [x] Modify the demo yjs server to connect based on room
- [x] Add an option to accept room code and join a particular room invite

### PR DESCRIPTION
**Pull request by adityathakurxd**
Create a new room [using Rooms API](https://www.100ms.live/docs/server-side/v2/api-reference/Rooms/create-via-api) and [Room Codes](https://www.100ms.live/docs/server-side/v2/api-reference/room-codes/create-room-code-api) for users to join individual rooms when trying out the demo.

Fixes #13

### CODE REVIEW

File: app/components/JoinForm.tsx [hunk 1]
```diff
@@ -41,14 +41,47 @@ const JoinForm = () => {
...
+			const response = await fetch('/api/create', {
+				method: 'GET',
+			})
+
+			response.json().then(async (result) => {
+				const responseBody = result.body
+				if (responseBody) {
+					const roomId = responseBody.id
+
+					const res = await fetch('/api/create', {
```
Comment (KaustubhKumar05) [inline app/components/JoinForm.tsx:53]: Better to stick with await and remove then at L48. 

```
const responseBody = await response.json()
// Rest goes here

File: app/components/JoinForm.tsx [hunk 2]
```diff
@@ -41,14 +41,47 @@ const JoinForm = () => {
...
+						method: 'POST',
+						headers: {
+							'Content-Type': 'application/json',
+						},
+						body: JSON.stringify({ roomId }),
+					})
+
+					res.json().then(async (result) => {
```
Comment (KaustubhKumar05) [inline app/components/JoinForm.tsx:61]: Same as above

File: app/components/JoinForm.tsx [hunk 3]
```diff
@@ -41,14 +41,47 @@ const JoinForm = () => {
...
+
+							var authToken = ''
+
+							if (activeTabRole === 'teacher') {
+								authToken = await hmsActions.getAuthTokenByRoomCode({
+									roomCode: roomCodeForTeacher,
+								})
+							} else {
+								authToken = await hmsActions.getAuthTokenByRoomCode({
+									roomCode: roomCodeForStudent,
```
Comment (KaustubhKumar05) [inline app/components/JoinForm.tsx:75]:
```
roomCode: condition? roomCodeForTeacher:roomCodeForStudent
````
works too

File: app/components/JoinForm.tsx [hunk 4]
```diff
@@ -41,14 +41,47 @@ const JoinForm = () => {
...
+						body: JSON.stringify({ roomId }),
+					})
+
+					res.json().then(async (result) => {
+						const data = result.body.data
+						if (data.length >= 2) {
+							const roomCodeForStudent = data[0].code
+							const roomCodeForTeacher = data[1].code
+
+							var authToken = ''
```
Comment (KaustubhKumar05) [inline app/components/JoinForm.tsx:67]: We should use `let`

File: app/components/JoinForm.tsx [hunk 5]
```diff
@@ -41,14 +41,47 @@ const JoinForm = () => {
 		const { name: userName = '', roomCode = '' } = inputValues
```
Comment (KaustubhKumar05) [inline app/components/JoinForm.tsx:41]: roomCode seems to be unused, we can remove it

File: app/components/JoinForm.tsx [hunk 6]
```diff
@@ -41,14 +41,47 @@ const JoinForm = () => {
...
+							await hmsActions.join({ userName, authToken })
+						}
+					})
+				} else {
+					console.error("Response body is empty or does not have an 'id' field.")
+				}
+			})
 		} catch (e) {
 			console.error(e)
 		}
```
Comment (KaustubhKumar05) [inline app/components/JoinForm.tsx:87]: We can probably show an error toast with some actionable message

**Comment by adityathakurxd** [comment]
To test this out, will need to create a `.env.local` with a `MANAGEMENT_TOKEN` from [100ms Dashboard](https://100ms.live).

**Comment by adityathakurxd** [comment]
Latest commit should add option to invite users to the demo. Would check for search params like `?room=` and the room code to join.

Need to fix the css for the button, improve the invite section and add a clipboard icon to copy button.

Screenshot 2024-02-29 at 1 46 48 PM

Screenshot 2024-02-29 at 1 47 06 PM

**Pull request merged**